srcStatic/OP2Memory.cpp

```diff
@@ -39,34 +41,51 @@ bool Op2MemCopy(void* destBaseAddr, void* sourceAddr, int size)
 	return (bSuccess != 0);
 }
 
+
+bool Op2MemSet(void* destBaseAddr, unsigned char value, int size)
+{
+	return Op2MemEdit(
+		destBaseAddr,
+		size,
+		[value](void* destAddr, std::size_t size) { memset(destAddr, value, size); }
+	);
+}
+
+bool Op2MemCopy(void* destBaseAddr, void* sourceAddr, int size)
+{
+	return Op2MemEdit(
+		destBaseAddr,
+		size,
+		[sourceAddr](void* destAddr, std::size_t size) { memcpy(destAddr, sourceAddr, size); }
+	);
+}
+
 bool Op2MemSetDword(void* destBaseAddr, int dword)
 {
-	// Chain to the memory copy function
 	return Op2MemCopy(destBaseAddr, &dword, sizeof(dword));
 }
 
 bool Op2MemSetDword(void* destBaseAddr, void* dword)
 {
-	return Op2MemSetDword(destBaseAddr, (int)dword);
+	return Op2MemCopy(destBaseAddr, &dword, sizeof(dword));
 }
 
-bool Op2MemSet(void* destBaseAddr, unsigned char value, int size)
+// This is used to patch up CALL instructions to intra-module non-virtual functions
+// The CALL instruction uses an encoding relative to the start of the next instruction.
+// Example:  E8 = CALL, 00040000 = relative offset = <DWORD(&someMethod - &postCallInstruction)>
+//   CALL someMethod  ; Encoded as E8 00040000
+//   postCallInstruction:
+// The `callOffset` parameter is the address of the encoded DWORD
+bool Op2RelinkCall(std::size_t callOffset, void* newFunctionAddress)
 {
-	void* destAddr = (void*)(loadOffset + (int)destBaseAddr);
+	const auto postCallInstructionAddress = loadOffset + callOffset + sizeof(void*);
+	return Op2MemSetDword(reinterpret_cast<void*>(callOffset), reinterpret_cast<std::size_t>(newFunctionAddress) - postCallInstructionAddress);
+}
 
+bool Op2UnprotectMemory(std::size_t destBaseAddr, std::size_t size)
```

Comment:
In OP2UnprotectMemory, the following error posting was removed. It seemed to me like maybe this is a mistake.

```C++
if (!bSuccess) {
	PostErrorMessage(__FILE__, __LINE__, "Error unprotecting memory at: " + std::to_string(reinterpret_cast<unsigned int>(destAddr)));
	return false;
}
```

Reply:
Ahh yes, correct. That was a mistake. That message should still be in `ExtInit` in srcDL/DllMain.cpp. The `Op2UnprotectMemory` method returns a `bool`, so that can be used to display the dialog. Ideally the dialog should be created in the higher level code.